pkg/services/ngalert/remote/alertmanager_test.go

```diff
@@ -143,8 +153,26 @@ func TestIntegrationRemoteAlertmanagerApplyConfigOnlyUploadsOnce(t *testing.T) {
 		OrgID:                     1,
 	}
 
+	silences := []byte("test-silences")
+	nflog := []byte("test-notifications")
+	store := fakes.NewFakeKVStore(t)
+	fstore := notifier.NewFileStore(1, store, "")
+
 	ctx := context.Background()
-	am, err := NewAlertmanager(cfg, 1, nil)
+	require.NoError(t, store.Set(ctx, 1, "alertmanager", notifier.SilencesFilename, base64.StdEncoding.EncodeToString(silences)))
+	require.NoError(t, store.Set(ctx, 1, "alertmanager", notifier.NotificationLogFilename, base64.StdEncoding.EncodeToString(nflog)))
+
+	fs := clusterpb.FullState{
+		Parts: []clusterpb.Part{
+			{Key: "silences", Data: silences},
+			{Key: "notifications", Data: nflog},
+		},
+	}
+	fullState, err := fs.Marshal()
+	require.NoError(t, err)
+	encodedFullState := base64.StdEncoding.EncodeToString(fullState)
+
+	am, err := NewAlertmanager(cfg, 1, fstore)
```

Comment:
you're missing a check right after this code - 

In the block that reads: `// We should have no configuration at first.` it should also be `// We should have no state or configuration at first.